src/document/range.js

```diff
@@ -72,6 +72,211 @@ CKEDITOR.define( [ 'document/positioniterator', 'document/position' ], ( Positio
 		}
 
 		/**
+		 * Checks whether this {document.Range range} intersects with the given {document.Range range}.
+		 *
+		 * @param {document.Range} otherRange Range to check.
+		 * @returns {boolean} True if ranges intersect.
+		 */
+		intersectsWith( otherRange ) {
```

Comment:
This is true. When combined with your other comment about changing `getCommon` to `getIntersection` it would also read natural:
`if ( a.getIntersection( b ) === null ) { }`.